hyperspy/tests/io/test_hdf5.py

```diff
@@ -365,6 +368,315 @@ def teardown_method(self, method):
+        m0 = markers.point(x=x1, y=y1)
+        m1 = markers.horizontal_line(y=y1)
+        m2 = markers.horizontal_line_segment(x1=x1, x2=x2, y=y1)
+        m3 = markers.line_segment(x1=x1, x2=x2, y1=y1, y2=y2)
+        m4 = markers.rectangle(x1=x1, x2=x2, y1=y1, y2=y2)
+        m5 = markers.text(x=x1, y=y1, text="test")
+        m6 = markers.vertical_line(x=x1)
+        m7 = markers.vertical_line_segment(x=x1, y1=y1, y2=y2)
+
+        s.add_marker(m0, permanent=True)
+        s.add_marker(m1, permanent=True)
+        s.add_marker(m2, permanent=True)
+        s.add_marker(m3, permanent=True)
+        s.add_marker(m4, permanent=True)
+        s.add_marker(m5, permanent=True)
+        s.add_marker(m6, permanent=True)
+        s.add_marker(m7, permanent=True)
+
+        with tempfile.TemporaryDirectory() as tmp:
+            filename = tmp + '/testallmarkersfile.hdf5'
+        s.save(filename)
+        s1 = load(filename)
+        m_dict = s1.metadata.Markers
+
+        m0_new_dict = sanitize_dict(m_dict.get_item(m0.name)._to_dictionary())
+        m1_new_dict = sanitize_dict(m_dict.get_item(m1.name)._to_dictionary())
+        m2_new_dict = sanitize_dict(m_dict.get_item(m2.name)._to_dictionary())
+        m3_new_dict = sanitize_dict(m_dict.get_item(m3.name)._to_dictionary())
+        m4_new_dict = sanitize_dict(m_dict.get_item(m4.name)._to_dictionary())
+        m5_new_dict = sanitize_dict(m_dict.get_item(m5.name)._to_dictionary())
+        m6_new_dict = sanitize_dict(m_dict.get_item(m6.name)._to_dictionary())
+        m7_new_dict = sanitize_dict(m_dict.get_item(m7.name)._to_dictionary())
+
+        assert len(list(s1.metadata.Markers)) == 8
+        assert sanitize_dict(m0._to_dictionary()) == m0_new_dict
+        assert sanitize_dict(m1._to_dictionary()) == m1_new_dict
+        assert sanitize_dict(m2._to_dictionary()) == m2_new_dict
+        assert sanitize_dict(m3._to_dictionary()) == m3_new_dict
+        assert sanitize_dict(m4._to_dictionary()) == m4_new_dict
+        assert sanitize_dict(m5._to_dictionary()) == m5_new_dict
+        assert sanitize_dict(m6._to_dictionary()) == m6_new_dict
+        assert sanitize_dict(m7._to_dictionary()) == m7_new_dict
+
+    def test_save_load_horizontal_line_marker(self):
+        y = 8
+        color = 'blue'
+        linewidth = 2.5
+        name = "horizontal_line_test"
+        s = Signal2D(np.arange(100).reshape(10, 10))
+        m = markers.horizontal_line(y=y, color=color, linewidth=linewidth)
+        m.name = name
+        s.add_marker(m, permanent=True)
+        with tempfile.TemporaryDirectory() as tmp:
+            filename = tmp + '/test_save_horizontal_line_marker.hdf5'
+        s.save(filename)
+        s1 = load(filename)
+        marker1 = s1.metadata.Markers.get_item(name)
+        assert marker1.name == name
+        assert marker1.get_data_position('y1') == y
+        assert marker1.marker_properties['color'] == color
+        assert marker1.marker_properties['linewidth'] == linewidth
```

Comment:
Also in all these tests multiple `assert` statements could all just be replaced by the sanitized dictionary comparison (which was why I suggested it in the first place..) - and you'd not need to carry that all relevant attributes are listed manually.